radis/io/hitran.py

```diff
@@ -215,6 +227,104 @@ def hit2df(
         if df is not None:
             return df
 
+    #  %% Start reading the full file
+
+    df = parse_hitran_file(fname, columns)
+
+    df = post_process_hitran_data(
+        df,
+        fname=fname,
+        parse_quanta=parse_quanta,
+    )
+    # cached file mode but cached file doesn't exist yet (else we had returned)
+    if cache:
+        new_metadata = {
+            # Last modification time of the original file :
+            "last_modification": time.ctime(getmtime(fname)),
+            "wavenum_min": df.wav.min(),
+            "wavenum_max": df.wav.max(),
+        }
+        if verbose:
+            print(
+                "Generating cache file {0} with metadata :\n{1}".format(
+                    fcache, new_metadata
+                )
+            )
+        from radis import __version__
+
+        try:
+            save_to_hdf(
+                df,
+                fcache,
+                metadata=new_metadata,
+                version=__version__,
+                overwrite=True,
+                verbose=verbose,
+                engine=engine,
+            )
+        except PermissionError:
+            if verbose:
+                print(sys.exc_info())
+                print("An error occured in cache file generation. Lookup access rights")
+            pass
+
+    # TODO : get only wavenum above/below 'load_wavenum_min', 'load_wavenum_max'
+    # by parsing df.wav.   Completely irrelevant files are discarded in 'load_h5_cache_file'
+    # but files that have partly relevant lines are fully loaded.
+    # Note : cache file is generated with the full line list.
+
+    return df
+
+
+def post_process_hitran_data(
+    df,
+    fname,
```

Comment:
we don't need to read the filename; you only use it to guess what's the molecule. Give it the molecule directly as an input parameter

Reply:
Okay, so then we would also have to pass molecule in the `hit2df` function. I'll have to add `molecule` as a parameter in the definition of `hit2df` as well then only `post_process_hitran_data` can be easily shared by `hit2df` and  `download_all_hitran_isotopes`

Reply:
I don't think so ; hit2df reads the file so it can infer it itself

Reply:
Okay